### PR TITLE
feat: download patch support for vanced.

### PR DESCRIPTION
**Pull request by Aunali321**
## TODO

- [ ] Compatibility with `com.vanced.android.youtube` for `settings` patch dependency

### CODE REVIEW

**Comment by oSumAtrIX** [comment]
Can you restrain the Vanced package to the latest version or the version you tested the patch on?

**Comment by Aunali321** [comment]
Yes sure and no settings don't work in vanced.

**Comment by oSumAtrIX** [comment]
Then the patch is not fully compatible with Vanced.

**Comment by Sculas** [comment]
What is the point of the downloads patch for Vanced if there are no settings? Users would be limited to NewPipe.
Is that acceptable? If so, it doesn't matter if it doesn't have settings.

File: src/main/kotlin/app/revanced/patches/youtube/misc/videoid/annotation/VideoIdCompatibility.kt [hunk 1]
```diff
@@ -4,9 +4,11 @@ import app.revanced.patcher.annotation.Compatibility
 import app.revanced.patcher.annotation.Package
 
 @Compatibility(
-    [Package(
-        "com.google.android.youtube", arrayOf("17.14.35", "17.22.36", "17.26.35", "17.27.39", "17.28.34", "17.29.34", "17.32.35")
-    )]
+    [
+        Package(
+        "com.google.android.youtube", arrayOf("17.14.35", "17.22.36", "17.26.35", "17.27.39", "17.28.34", "17.29.34", "17.32.35")),
+        Package("com.vanced.android.youtube", arrayOf("17.03.38"))
```
[the remark this answers is not in the document]
Comment (Aunali321) [reply src/main/kotlin/app/revanced/patches/youtube/misc/videoid/annotation/VideoIdCompatibility.kt:10]: the patch itself succeeds and works except for settings. you cant change package name. but the newpipe intent launch and downloading functionality works.

**Comment by oSumAtrIX** [comment]
To make this patch properly compatible with `com.vanced.android.youtube`, the patches dependencies should succeed too, such as the `settings` patch. If this can not be guaranteed, this PR should be closed and re-opened once this issue is solved.

**Comment by Sculas** [comment]
If @Aunali321 wants to continue on this PR, it should be converted to a draft, not closed. @oSumAtrIX 
Only if they don't want to continue should we close the PR :)

**Comment by Sculas** [comment]
The settings patch needs to work on Vanced. If it doesn't, support for it needs to be added.
If that's done, this PR can be merged.

**Comment by oSumAtrIX** [comment]
A `TODO` section in the PR description has been added for that.

**Comment by KrystianQur** [comment]
I tried to patch it, but i got that error:
```
SEVERE: downloads failed:
app.revanced.patcher.patch.PatchResultError: 'downloads' depends on 'downloads' but the following error was raised: 'downloads-resource-patch' depends on 'downloads-resource-patch' but the following error was raised: java.lang.NullPointerException
        at app.revanced.patcher.Patcher.applyPatch(Patcher.kt:259)
        at app.revanced.patcher.Patcher.access$applyPatch(Patcher.kt:46)
        at app.revanced.patcher.Patcher$applyPatches$1.invokeSuspend(Patcher.kt:383)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlin.sequences.SequenceBuilderIterator.hasNext(SequenceBuilder.kt:127)
        at app.revanced.utils.patcher.PatcherKt.applyPatchesVerbose(Patcher.kt:92)
        at app.revanced.cli.patcher.Patcher.start$revanced_cli(Patcher.kt:24)
        at app.revanced.cli.command.MainCommand.run(MainCommand.kt:157)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1939)
        at picocli.CommandLine.access$1300(CommandLine.java:145)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2358)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2352)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2314)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2316)
        at picocli.CommandLine.execute(CommandLine.java:2078)
        at app.revanced.cli.main.MainKt.main(Main.kt:7)
Exception in thread "main" java.lang.NoSuchMethodError: 'app.revanced.patcher.fingerprint.method.impl.PatternScanResult app.revanced.patcher.fingerprint.method.impl.MethodFingerprintResult.getPatternScanResult()'
        at app.revanced.patches.youtube.misc.clientspoof.patch.ClientSpoofPatch.execute(ClientSpoofPatch.kt:30)
        at app.revanced.patches.youtube.misc.clientspoof.patch.ClientSpoofPatch.execute(ClientSpoofPatch.kt:18)
        at app.revanced.patcher.Patcher.applyPatch(Patcher.kt:285)
        at app.revanced.patcher.Patcher.access$applyPatch(Patcher.kt:46)
        at app.revanced.patcher.Patcher$applyPatches$1.invokeSuspend(Patcher.kt:383)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlin.sequences.SequenceBuilderIterator.hasNext(SequenceBuilder.kt:127)
        at app.revanced.utils.patcher.PatcherKt.applyPatchesVerbose(Patcher.kt:92)
        at app.revanced.cli.patcher.Patcher.start$revanced_cli(Patcher.kt:24)
        at app.revanced.cli.command.MainCommand.run(MainCommand.kt:157)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1939)
        at picocli.CommandLine.access$1300(CommandLine.java:145)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2358)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2352)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2314)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2316)
        at picocli.CommandLine.execute(CommandLine.java:2078)
        at app.revanced.cli.main.MainKt.main(Main.kt:7)
```

**Pull request closed without merging**